build/release/teamcity-publish-release.sh

```diff
@@ -1,4 +1,144 @@
 #!/usr/bin/env bash
 
-echo "Implement me!"
+set -euxo pipefail
 
+source "$(dirname "${0}")/teamcity-support.sh"
+
+
+tc_start_block "Variable Setup"
+export BUILDER_HIDE_GOPATH_SRC=1
+
+# Matching the version name regex from within the cockroach code except
+# for the `metadata` part at the end because Docker tags don't support
+# `+` in the tag name.
+# https://github.com/cockroachdb/cockroach/blob/master/pkg/util/version/version.go#L75
+build_name="$(echo "${NAME}" | grep -E -o '^v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-[-.0-9A-Za-z]+)?$')"
```

Comment:
I tried using `+dryrun` instead of `.dryrun` or `-dryrun` and found docker image tags don't allow `+` in the tag name. I chose to not allow `+` in the build name so this problem isn't encountered. This means we can't have releases (or even internal builds that we want CC to consume) with `+`'s in them. I'm open to alternative ways of solving this if you've any ideas.